ui/pages/confirmation/confirmation.js

```diff
@@ -258,14 +262,24 @@ export default function ConfirmationPage({
     return INPUT_STATE_CONFIRMATIONS.includes(type);
   };
 
-  const handleSubmit = () =>
-    templateState[pendingConfirmation.id]?.useWarningModal
+  const handleSubmit = async () => {
+    setLoading(true);
+    const submitResult = await (templateState[pendingConfirmation.id]
```

Comment:
Nit: Sorry I know I requested that the inner `await` be removed earlier. But I hadn't considered which we needed here, I just noticed the redundancy. On second look, it does seem to make more sense for the `await` to just be used for `onSubmit`, since `useWarningModal` is a boolean property not a Promise

Reply:
Could you remove this outer `await`, now that the inner one has been restored?